File: _fix.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'configPanel.js');
let content = fs.readFileSync(filePath, 'utf8');

// The problem: after </html> there should be a backtick + semicolon to close the template literal,
// then a closing brace for getWebviewContent function.
// Currently it's: </html>\r\n}; or </html>\n};
// We need:       </html>` + ;\r\n} or </html>` + ;\n}

const bt = String.fromCharCode(96); // backtick character

// Try both line ending styles
const patterns = [
    { old: '</html>\r\n};', newStr: '</html>' + bt + ';\r\n}' },
    { old: '</html>\n};', newStr: '</html>' + bt + ';\n}' },
    { old: '</html>\r\n\r\n/**', newStr: '</html>' + bt + ';\r\n}\r\n\r\n/**' },
    { old: '</html>\n\n/**', newStr: '</html>' + bt + ';\n}\n\n/**' },
];

let fixed = false;
for (const p of patterns) {
    if (content.includes(p.old)) {
        content = content.replace(p.old, p.newStr);
        fs.writeFileSync(filePath, content, 'utf8');
        fs.writeFileSync(path.join(__dirname, '_fix_result.txt'), 'Fixed with pattern: ' + JSON.stringify(p.old), 'utf8');
        fixed = true;
        break;
    }
}

if (!fixed) {
    // Dump what's around </html> for debugging
    const idx = content.indexOf('</html>');
    let debug = 'Not fixed. idx=' + idx + '\n';
    if (idx >= 0) {
        const after = content.substring(idx, idx + 30);
        debug += 'After: ' + JSON.stringify(after) + '\n';
        debug += 'Has backtick after html: ' + (content.charAt(idx + 7) === bt) + '\n';
    }
    fs.writeFileSync(path.join(__dirname, '_fix_result.txt'), debug, 'utf8');
}

File: _fix.py
import os

f = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'src', 'configPanel.js')
with open(f, 'r', encoding='utf-8') as fh:
    c = fh.read()

idx = c.find('</html>')
debug = f'idx={idx}\n'
if idx >= 0:
    after = c[idx+7:idx+17]
    debug += f'after={repr(after)}\n'

bt = chr(96)  # backtick

# Try CRLF
old1 = '</html>\r\n};'
new1 = '</html>' + bt + ';\r\n}'
# Try LF
old2 = '</html>\n};'
new2 = '</html>' + bt + ';\n}'

if old1 in c:
    c = c.replace(old1, new1, 1)
    with open(f, 'w', encoding='utf-8') as fh:
        fh.write(c)
    debug += 'FIXED_CRLF\n'
elif old2 in c:
    c = c.replace(old2, new2, 1)
    with open(f, 'w', encoding='utf-8') as fh:
        fh.write(c)
    debug += 'FIXED_LF\n'
else:
    debug += 'NO_MATCH\n'
    debug += f'has_backtick_after_html={bt in c[idx+7:idx+10]}\n'

result_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), '_fix_result.txt')
with open(result_path, 'w', encoding='utf-8') as fh:
    fh.write(debug)
